### ISSUE TITLE
Allow omitting location/track-caller via function

### ISSUE THREAD

**Issue by Emilgardis** [yaahc/color-eyre#105]
Add a `HookBuilder` method that will enable or disable output of tracked location.

e.g

```rust
use color_eyre::config::HookBuilder;
fn main() -> eyre::Result<()> {
    HookBuilder::new().display_location(false).install()?;
    eyre::bail!("an error");
}
```

gives

```text
Error:
   0: an error

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
```

**Comment by yaahc** [comment]
Right now this is supported via a feature: https://github.com/yaahc/color-eyre/blob/master/Cargo.toml#L15

**Comment by Emilgardis** [comment]
Indeed! However, I want to conditionally enable location output in a application. This would be chosen at runtime with a environment variable.

so

```rust
use color_eyre::config::HookBuilder;
fn main() -> eyre::Result<()> {
    HookBuilder::new().display_location(std::env::var("DISPLAY_LOCATION").is_ok()).install()?;
    eyre::bail!("an error");
}
```

**Comment by yaahc** [comment]
> Indeed! However, I want to conditionally enable location output in a application. This would be chosen at runtime with a environment variable.
> 
> so
> 
> ```rust
> use color_eyre::config::HookBuilder;
> fn main() -> eyre::Result<()> {
>     HookBuilder::new().display_location(std::env::var("DISPLAY_LOCATION").is_ok()).install()?;
>     eyre::bail!("an error");
> }
> ```

ah, okay, that's fair enough. I'd be happy to accept a PR adding a method for configuring this at runtime, though for consistency let's call it `display_location_section`.
